client/web/src/search/stream.tsx

```diff
@@ -64,6 +65,63 @@ interface CommitMatch {
 
 type RepositoryMatch = Pick<FileMatch, 'repository' | 'branches'>
 
+// Aggregate type. Replace when new ones come in.
+interface Progress {
+    /**
+     * True if this is the final progress update for this stream
+     */
+    done: boolean
+
+    // The number of repositories matching the repo: filter. Is set once they
+    // are resolved.
+    repositoriesCount?: number
+
+    // The number of non-overlapping matches. If skipped is non-empty, then
+    // this is a lower bound.
+    matchCount: number
+
+    // Wall clock time in milliseconds for this search.
+    durationMS: number
+
+    // A description of shards or documents that were skipped. This has a
+    // deterministic ordering.
+    skipped: Skipped[]
+}
+
+interface Skipped {
+    /**
+     * Why a document/shard/repository was skipped. We group counts by reason.
+     *
+     * - document-match-limit :: we found too many matches in a document, so we stopped searching it.
+     * - shard-match-limit :: we found too many matches in a shard/repository, so we stopped searching it.
+     * - shard-timeout :: we ran out of time before searching a shard/repository.
+     * - repository-cloning :: we could not search a repository because it is not cloned.
+     * - repository-missing :: we could not search a repository because it is not cloned and we failed to find it on the remote code host.
+     * - excluded-fork :: we did not search a repository because it is a fork.
+     * - excluded-archive :: we did not search a repository because it is archived.
+     * - repository-limit :: we did not search a repository because the set of repositories to search was too large.
+     */
+    reason:
+        | 'document-match-limit'
+        | 'shard-match-limit'
+        | 'shard-timedout'
+        | 'repository-cloning'
+        | 'repository-missing'
+        | 'excluded-fork'
+        | 'excluded-archive'
+        | 'repository-limit'
+    // A short message. eg 1,200 timedout.
+    title: string
+    // A message to show the user. Usually includes information explaining the reason, count as well as a sample of the missing items.
+    message: string
+    severity: 'info' | 'warn'
+    // a suggested query expression to remedy the skip. eg "archived:yes" or "timeout:2m".
+    suggested?: {
+        title: string
+        queryExpression: string
+    }
```

Comment:
```suggestion
    /**
     * A short message. eg 1,200 timed out.
     */
    title: string
    /**
     * A message to show the user. Usually includes information explaining the reason,
     * count as well as a sample of the missing items.
     */
    message: string
    severity: 'info' | 'warn'
    /**
     * a suggested query expression to remedy the skip. eg "archived:yes" or "timeout:2m".
     */
    suggested?: {
        title: string
        queryExpression: string
    }
```

Let's have docstrings everywhere :)

Reply:
Love the java influence :D